rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -859,6 +855,55 @@ public void onSuccess(String result) {
         });
     }
 
+    /**
+     * Fetch and store in the model the current configuration of a node source
+     *
+     * @param success call this when it's done
+     * @param failure call this if it fails
+     */
+    public void fetchNodeSourceConfiguration(String nodeSourceName, Runnable success, Runnable failure) {
+
+        rm.getNodeSourceConfiguration(LoginModel.getInstance().getSessionId(),
+                                      nodeSourceName,
+                                      new AsyncCallback<String>() {
+
+                                          public void onSuccess(String result) {
+                                              model.setEditedNodeSourceConfiguration(parseNodeSourceConfiguration(result));
+                                              success.run();
+                                          }
+
+                                          public void onFailure(Throwable caught) {
+                                              String msg = JSONUtils.getJsonErrorMessage(caught);
+                                              SC.warn("Failed to fetch configuration of node source " + nodeSourceName +
+                                                      ":<br>" + msg);
+                                              failure.run();
+                                          }
+                                      });
+    }
+
+    private NodeSourceConfiguration parseNodeSourceConfiguration(String json) {
+
+        JSONObject jsonObject = this.parseJSON(json).isObject();
+
+        String nodeSourceName = jsonObject.get("nodeSourceName").isString().stringValue();
+
+        boolean nodesRecoverable = jsonObject.get("nodesRecoverable").isBoolean().booleanValue();
+
+        JSONObject infrastructurePluginDescriptorJson = jsonObject.get("infrastructurePluginDescriptor").isObject();
+        String infrastructurePluginName = infrastructurePluginDescriptorJson.get("pluginName").isString().stringValue();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "pluginName" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)